core/kazoo_number_manager/src/knm_phone_number.erl

```diff
@@ -95,7 +95,8 @@ new(DID, Options) ->
                 ,{fun set_number_db/2, knm_converters:to_db(NormalizedNum)}
                 ,{fun set_assign_to/2, knm_number_options:assign_to(Options)}
                 ,{fun set_state/2, ?NUMBER_STATE_DISCOVERY}
-                ,{fun set_module_name/2, knm_number_options:module_name(Options)}
+                ,{fun set_module_name/2, ?CARRIER_OTHER}
```

Comment:
why this revert ? and why `?CARRIER_OTHER` ? the system can be configured not to use this carrier, right ?
if not available in options, shouldn't we get it from config (maybe `available_module_name`) ?